userspace/libsinsp/container_engine/docker/connection_linux.cpp

```diff
@@ -14,113 +14,94 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 */
+#include "connection.h"
 
-#include "container_engine/docker.h"
-
-#include "runc.h"
-#include "container_engine/mesos.h"
 #include "sinsp.h"
 #include "sinsp_int.h"
 
-using namespace libsinsp::container_engine;
-using namespace libsinsp::runc;
-
 namespace {
 
-size_t docker_curl_write_callback(const char* ptr, size_t size, size_t nmemb, string* json)
+size_t docker_curl_write_callback(const char *ptr, size_t size, size_t nmemb, std::string *json)
 {
 	const std::size_t total = size * nmemb;
 	json->append(ptr, total);
 	return total;
 }
 
-constexpr const cgroup_layout DOCKER_CGROUP_LAYOUT[] = {
-	{"/", ""}, // non-systemd docker
-	{"/docker-", ".scope"}, // systemd docker
-	{nullptr, nullptr}
-};
 }
 
-std::string docker::m_docker_sock = "/var/run/docker.sock";
+using namespace libsinsp::container_engine;
 
-void docker::cleanup()
+docker_connection::docker_connection():
+	m_api_version("/v1.24"),
+	m_curlm(nullptr)
 {
-	m_docker_info_source.reset(NULL);
-}
+	m_curlm = curl_multi_init();
 
-void docker_async_source::init_docker_conn()
-{
-	if(!m_curlm)
+	if(m_curlm)
 	{
-		m_curl = curl_easy_init();
```

Comment:
Why was m_curl dropped in favor of get_docker() local curl? 
I mean, what is the rationale behind the choice?

Reply:
Each curl instance uses a single Unix socket path (see line 66 below). We could keep switching that (podman can talk to a number of different sockets, one per user) but IIRC I ran into concurrency issues (requests going to the wrong socket) so I took the easy way out.

Reply:
That makes sense, thanks!